scripts/devops_tasks/install_python_version.py

```diff
@@ -79,38 +75,71 @@ def install_selected_python_version(installer_url, installer_folder):
+    target = Version(requested_version)
+
+    if len(requested_version.split(".")) > 2:
+        return requested_version
+    else:
+        target_versions = [
+            Version(version) for version in version_manifest.keys() if version.startswith(requested_version)
+        ]
+
+        if target_versions:
+            return target_versions[0]
+        else:
+            print(f'Unable to select a valid version from manifest for version "{requested_version}"')
+
+
 def get_installer_url(requested_version, version_manifest):
     current_plat = platform.system().lower()
-
     print("Current Platform Is {}".format(platform.platform()))
 
-    if version_manifest[requested_version]:
-        found_installers = version_manifest[requested_version]["files"]
+    actual_requested_version = get_resolvable_version(requested_version, version_manifest)
+
+    if actual_requested_version in version_manifest:
+        found_installers = version_manifest[actual_requested_version]["files"]
 
         # filter anything that's not x64. we don't care.
-        x64_installers = [
-            file_def for file_def in found_installers if file_def["arch"] == "x64"
-        ]
+        x64_installers = [file_def for file_def in found_installers if file_def["arch"] == "x64"]
 
         if current_plat == "windows":
-            return [
-                installer
-                for installer in x64_installers
-                if installer["platform"] == "win32"
-            ][0]
+            return [installer for installer in x64_installers if installer["platform"] == "win32"][0]
         elif current_plat == "darwin":
-            return [
-                installer
-                for installer in x64_installers
-                if installer["platform"] == current_plat
-            ][0]
+            return [installer for installer in x64_installers if installer["platform"] == current_plat][0]
         else:
             return [
                 installer
                 for installer in x64_installers
-                if installer["platform"] == "linux"
-                and installer["platform_version"] == CURRENT_UBUNTU_VERSION
+                if installer["platform"] == "linux" and installer["platform_version"] == CURRENT_UBUNTU_VERSION
             ][0]
+    else:
+        print(
+            f"Requested version {actual_requested_version} is not available from the manifest at {MANIFEST_LOCATION}."
+        )
+
+
+def necessary_to_install(version_requested) -> bool:
+    version_from_spec = Version(version_requested)
+    precached_version = Version(MAX_PRECACHED_VERSION)
+    precached = True
+
+    # the azure devops folks would prefer that we request python version by Major.Minor, EG
+    # "3.11", "3.9". However, in cases like "3.11", the max precached version is "3.11.1".
+    # This means that if we do a simple check of 3.11 > 3.11.1, we will FAIL, which will result in
+    # unnecessarily installation.
```

Comment:
This is kind of confusing to me because it implies that 3.11 > 3.11.1 will have a different result than 3.11 > 3.11. Is there a missing equals sign in our condition below? Otherwise, I think the truncation of 3.11.1 is unnecessary
```suggestion
    # This means that if we do a simple check of 3.11 > 3.11.1, we will FAIL, which will result in
    # unnecessarily installation.
```

Reply:
Input is going to be `3.11` from OS. What we're trying to avoid is unnecessary installation. I'm just going to rework that entire comment to be more accurate. You're definitely not weird to be confused by this.

Reply:
Hey @mccoyp  I have updated this comment. I think it'll be more clear for you now.